### PR TITLE
Added vast majority of Gary's suggested visual changes

### PR DESCRIPTION
**Pull request by jonahdlin**
In this PR I modified a lot of visual attributes as per Gary's suggestions including

- New layout dropdown menu
  - I could have used a dependency for this but none of the well-documented and standalone ones I could find would do what I wanted
  - Now displays information about each layout in the dropdown menu
- Redid download cards in the sidebar
  - The entire card is now clickable. To make sure this is clear to the user, it scales a little on hover
  - Added react-loader loading icon to downloads (I couldn't figure out how to have it work properly with the png download though)
- Removed cola.js layout option, since it's very similar to CoSE-Bilkent and it may cause confusion to have two force-directed layouts
- Source name in info menu now links to datasource pages
- Small wording changes

As is the case for a lot of my PRs nowadays the code isn't too pretty, but we want to get a working demo with all of Jeff and Gary's aesthetic suggestions implemented so we can send it out to the whole Pathway Commons Dev team.

### CODE REVIEW

File: src/client/features/view/components/menu/index.js [hunk 1]
```diff
@@ -3,6 +3,11 @@ const h = require('react-hyperscript');
 const Link = require('react-router-dom').Link;
 const classNames = require('classnames');
 
+const layoutConf = require('../../../../common/cy/layout');
+
+const Dropdown = require('../../../../common/dropdown').Dropdown;
```
Comment (d2fong) [inline src/client/features/view/components/menu/index.js]: Pretty sure you can do this:
```js
const { Dropdown, DropdownOption } = require('../../../../common/dropdown');
```

File: src/client/features/view/components/sidebar/menus/components/downloadOption.js [hunk 2]
```diff
@@ -23,7 +24,11 @@ class DownloadOption extends React.Component {
     };
   }
 
-  handleDownloadClick(type) {
+  handleDownloadClick(evt, type) {
+    // Don't do anything when a link is clicked. Should not download just because
+    // user clicks a link
+    if (evt.target.tagName === 'A') return;
```
Comment (d2fong) [inline src/client/features/view/components/sidebar/menus/components/downloadOption.js]: Try:
```js
evt.preventDefault();
```
Instead of checking for the tagName and see if it works.

File: src/styles/features/view/animations/loadingSpin.css [hunk 3]
```diff
@@ -0,0 +1,9 @@
+@keyframes loading-spin {
```
Comment (d2fong) [inline src/styles/features/view/animations/loadingSpin.css]: Is this loading spinner specifically for the .png download that react-loader didn't work with?

Can you provide more detail about how react-loader doesn't work with the .png download?

File: src/styles/index.css [hunk 4]
```diff
@@ -11,3 +11,4 @@
 @import "./features/paint.css";
 @import "./features/view.css";
 @import "./features/admin.css";
+@import "./common/index.css";
```
Comment (d2fong) [inline src/styles/index.css]: common css should be after app.css but before features

File: src/styles/vars.css [hunk 5]
```diff
@@ -4,11 +4,13 @@
...
   --cloud: #ECF0F1;
   --cloud-dark: #d3d8da;
+  --cloud-darker: #b5b9bb;
 
   /* Colours used by other modules */
   --accent-1-colour: var(--green-sea);
   --light-base-colour: var(--cloud);
   --light-base-colour-dark: var(--cloud-dark);
+  --light-base-colour-darker: var(--cloud-darker);
   --dark-base-colour: var(--midnight-blue);
```
Comment (d2fong) [inline src/styles/vars.css]: @maxkfranz is this what you had in mind for semantic css var names?  I find it more difficult to remember ```accent-1-colour``` vs ```green-sea```.

File: src/styles/features/view/animations/loadingSpin.css (hunk 3 above)
Comment (jonahdlin) [reply src/styles/features/view/animations/loadingSpin.css]: The loading spinner CSS I left in by mistake before I realized react-loader would work fine.

As for the .png download, unlike the other formats, comes straight from cytoscape.js, and is then saved with saveAs from file-saver. I couldn't figure out how to tell when the procedure is done since neither saveAs nor cy.png() return promises or take callbacks.

File: src/client/features/view/components/menu/index.js (hunk 1 above)
Comment (jonahdlin) [reply src/client/features/view/components/menu/index.js]: Yup, that works. I've made that change and I'll keep in mind to change others when I come across them.

File: src/styles/features/view/animations/loadingSpin.css (hunk 3 above)
Comment (d2fong) [reply src/styles/features/view/animations/loadingSpin.css]: Have you tried something like this already?
```js
this.setState({loading: true});
cy.png(...);
saveAs(blob, ..., ...);
this.setState({loading: false});
```

Comment (d2fong) [reply src/styles/features/view/animations/loadingSpin.css]: You can also do:

```js
const saveImage = new Promise((resolve, reject) => {
   const img = cy.png();
   saveAs(img, ...., ...);
   resolve(null);
});

saveImage.then(() => this.setState({loading: false});

```

Comment (d2fong) [reply src/styles/features/view/animations/loadingSpin.css]: It might be that saveAs the problem

File: src/client/features/view/components/sidebar/menus/components/downloadOption.js (hunk 2 above)
Comment (jonahdlin) [reply src/client/features/view/components/sidebar/menus/components/downloadOption.js]: `evt.preventDefault()` doesn't work in this case it would seem because it tries to prevent the default behaviour of the event. In this case I want the default behaviour and not the behaviour I define, so if a user clicks a link inside the download option, the link opens, but the download does not happen.

Comment (d2fong) [reply src/client/features/view/components/sidebar/menus/components/downloadOption.js]: So when a user clicks an 'a' tag, you want to open the link and download a file?

Comment (jonahdlin) [reply src/client/features/view/components/sidebar/menus/components/downloadOption.js]: No, I want to open the link and not download the file.

Comment (d2fong) [reply src/client/features/view/components/sidebar/menus/components/downloadOption.js]: Do you always want to open the link and not download the file or does that depend on the type of file you are downloading?

Comment (jonahdlin) [reply src/client/features/view/components/sidebar/menus/components/downloadOption.js]: Doesn't matter the file type. The links are clickable without downloading the file. The whole card is clickable for the download except the links.

File: src/styles/features/view/animations/loadingSpin.css (hunk 3 above)
Comment (jonahdlin) [reply src/styles/features/view/animations/loadingSpin.css]: I'll give that a shot tomorrow with some pathways.

Comment (jonahdlin) [reply src/styles/features/view/animations/loadingSpin.css]: That code runs into the same problems as the previous code. I've done more testing on it and it seems to be a synchronous process, that for some reason holds the entire page from rendering while executing. If we do
```js
this.setState({loading: true});
cy.png(...);
saveAs(blob, ..., ...);
this.setState({loading: false});
```
then the loader is correctly set to loading before the process executes, and stops loading afterwards, but never displays while it's active.

@hmtinc suggested wrapping the process in a `setTimeout`. This works as long as it's called as a callback to the first `setState`, i.e.
```js
this.setState({ loading: true }, () => {
  setTimeout(() => {
    saveAs(this.props.cy.png({
      ...
    }), ...);
    this.setState({ loading: false });
  }, 1);
});
```
Doesn't matter how long the timeout is (1ms works fine), and it has to be wrapped in the callback for `this.setState`.

File: src/client/services/cdc/index.js [hunk 6]
```diff
@@ -1,11 +1,21 @@
 const io = require('socket.io-client');
-let socket = io.connect('/');
+let socket = io.connect('192.168.81.233:3000');
```
Comment (d2fong) [inline src/client/services/cdc/index.js]: Revert this change
